com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -455,8 +456,31 @@ UnityInputProto Exchange(UnityOutputProto unityOutput)
                 QuitCommandReceived?.Invoke();
                 return message.UnityInput;
             }
+            catch (RpcException rpcException)
+            {
+                // Log more verbose errors if they're something the user can possibly do something about.
+                switch (rpcException.Status.StatusCode)
+                {
+                    case StatusCode.Unavailable:
+                        // This can happen when python disconnects. Ignore it to avoid noisy logs.
+                        break;
+                    case StatusCode.ResourceExhausted:
+                        // This happens is the message body is too large. There's no way to
+                        // gracefully handle this, but at least we can show the message and the
+                        // user can try to reduce the number of agents or observation sizes.
+                        Debug.LogError($"GRPC Exception: {rpcException.Message}. Exiting communicator.");
+                        break;
+                    default:
+                        // Other unknown errors. Keep these quiet for now.
+                        break;
```

Comment:
I could be convinced to log other exceptions here (maybe as Info instead of Error). That would at least help us for future unknown unknowns.